includes/class-newspack-popups-newsletters.php

```diff
@@ -153,60 +102,47 @@ private static function fetch_reader_data_from_esp( $email_address ) {
 		}
 
 		$nonce = \wp_create_nonce( 'newspack_campaigns_lightweight_api' );
-		$api   = Campaign_Data_Utils::get_api( $nonce );
-		if ( ! $api ) {
-			return;
-		}
 
-		$client_id           = Newspack_Popups_Segmentation::get_client_id();
 		$events_to_add       = [];
 		$newsletter_provider = \Newspack_Newsletters::get_service_provider();
 
-		// If the reader is already known to be a newsletter subscriber, no need to proceed.
-		$has_subscription_events = ! empty( $api->get_reader_events( $client_id, 'subscription', $email_address ) );
-		if ( ! $has_subscription_events ) {
-			// Look up the email address as a contact with the connected ESP. If not a contact, no need to proceed.
-			$subscribed_lists = \Newspack_Newsletters_Subscription::get_contact_lists( $email_address );
-			if ( ! is_wp_error( $subscribed_lists ) && ! empty( $subscribed_lists ) && is_array( $subscribed_lists ) ) {
-				// The reader is subscribed to one or more lists, so they should be segmented as a subscriber.
+		// Look up the email address as a contact with the connected ESP. If not a contact, no need to proceed.
+		$subscribed_lists = \Newspack_Newsletters_Subscription::get_contact_lists( $email_address );
+		if ( ! is_wp_error( $subscribed_lists ) && ! empty( $subscribed_lists ) && is_array( $subscribed_lists ) ) {
+			// The reader is subscribed to one or more lists, so they should be segmented as a subscriber.
+			$events_to_add[] = [
+				'type'    => 'subscription',
+				'context' => $email_address,
+				'value'   => [
+					'esp'   => $newsletter_provider->service,
+					'lists' => $subscribed_lists,
+				],
+			];
+
+			// TODO: set subscription status in the Reader Data store.
```

Comment:
Also resolved in https://github.com/Automattic/newspack-plugin/pull/2539

Reply:
You're right, thanks for pointing it out!

Reply:
Removed this file in 6c97c16